src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIMeleeGuard.java

```diff
@@ -129,8 +133,31 @@ protected AIState huntDown()
             targetEntity = this.worker.getLastAttacker();
         }
 
+        final AbstractEntityBarbarian barbarian = BarbarianUtils.getClosestBarbarianToEntity(worker, 20D);
+
+        if (barbarian != null)
+        {
+            targetEntity = barbarian;
+        }
+
+        if (targetEntity != null && worker.getColony() != null)
+        {
+            List<EntityLivingBase> targets = worker.getColony().getGuardTargets();
+            if (targets.stream().noneMatch(entity -> entity == targetEntity))
+            {
+                targets.add(targetEntity);
+            }
+            worker.getColony().setGuardTargets(targets);
+        }
+
         if (!targetEntity.isEntityAlive() || checkForToolOrWeapon(ToolType.SWORD))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "targetEntity" is nullable here [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)